clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -764,9 +765,13 @@ public void setAcknowledgementCommitCallback(final AcknowledgementCommitCallback
         try {
             if (callback != null) {
                 acknowledgementCommitCallbackHandler = new AcknowledgementCommitCallbackHandler(callback);
+                ShareAcknowledgeCommitCallbackHandlerEvent event = new ShareAcknowledgeCommitCallbackHandlerEvent(true);
```

Comment:
You only really need to send an event is there is a transition from no callback to callback, or callback to no callback. If someone registered a replacement callback, there's no need for the event. Similarly, if they unregistered a callback when actually there was no callback registered.

Reply:
Right yeah makes sense, have updated the PR.